api/envoy/config/retry_policy/conditional_retry/v2/conditional_retry.proto

```diff
@@ -0,0 +1,92 @@
+syntax = "proto3";
+
+package envoy.config.retry_policy.conditional_retry.v2;
+
+import "envoy/api/v2/route/route_components.proto";
+
+import "udpa/annotations/migrate.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.config.retry_policy.conditional_retry.v2";
+option java_outer_classname = "ConditionalRetryProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_migrate).move_to_package =
+    "envoy.extensions.retry_policy.conditional_retry.v3";
+
+// [#not-implemented-hide:]
+// Retry policy match configuration. This is a recursive structure which allows
+// complex nested match configurations to be built using various logical
+// operators.
+// [#next-free-field: 7]
+message MatchPredicate {
+  // A set of match configurations used for logical operations.
+  message MatchSet {
+    // The list of rules that make up the set.
+    repeated MatchPredicate rules = 1 [(validate.rules).repeated = {min_items: 2}];
+  }
+
+  oneof rule {
+    option (validate.required) = true;
+
+    // A set that describes a logical OR. If any member of the set matches,
+    // the match configuration matches.
+    MatchSet or_match = 1;
+
+    // A set that describes a logical AND. If all members of the set match,
+    // the match configuration matches.
+    MatchSet and_match = 2;
+
+    // A negation match. The match configuration will match if the negated
+    // match condition matches.
+    MatchPredicate not_match = 3;
+
+    // The match configuration will always match.
+    bool any_match = 4 [(validate.rules).bool = {const: true}];
+
+    // HTTP request header match configuration.
+    api.v2.route.HeaderMatcher http_request_header_match = 5;
+
+    // HTTP request trailer match configuration.
+    api.v2.route.HeaderMatcher http_request_trailer_match = 6;
```

Comment:
Out of curiosity what would this actually be used for?

Reply:
Maybe I don't need the trailer match now. The initial goal is to match the retry policy by request method.